pkg/platform/runtime/setup/setup.go

```diff
@@ -160,6 +163,98 @@ func NewWithModel(target Targeter, msgHandler Events, model ModelProvider, an an
 
 // Update installs the runtime locally (or updates it if it's already partially installed)
 func (s *Setup) Update() error {
+	var artifacts []artifact.ArtifactID
+	mutex := sync.Mutex{}
```

Comment:
```suggestion
	mutex := &sync.Mutex{}
```

Looking at the source of sync.Mutex it looks like its methods are expecting a pointer.